src/builder/scripts/TransferBuilderScripts.sol

```diff
@@ -0,0 +1,103 @@
+    }
+
+    function transfer(
+        TransferIntent memory transferIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory) {
+        // If the action is paid for with tokens, filter out any chain accounts that do not have corresponding payment information
+        if (payment.isToken) {
+            chainAccountsList = Accounts.findChainAccountsWithPaymentInfo(chainAccountsList, payment);
+        }
+
+        // Initialize TransferMax flag
+        bool isMaxTransfer = transferIntent.amount == type(uint256).max;
+        // TransferMax will always use quotecall to avoid leaving dust in wallet
+        bool useQuotecall = isMaxTransfer;
+
+        // Convert transferIntent to user aggregated balance
+        if (isMaxTransfer) {
+            transferIntent.amount = Accounts.totalAvailableAsset(transferIntent.assetSymbol, chainAccountsList, payment);
+        }
+
+        // Then, transfer `amount` of `assetSymbol` to `recipient`
+        (IQuarkWallet.QuarkOperation memory operation, Actions.Action memory action) = Actions.transferAsset(
+            Actions.TransferAsset({
+                chainAccountsList: chainAccountsList,
+                assetSymbol: transferIntent.assetSymbol,
+                amount: transferIntent.amount,
+                chainId: transferIntent.chainId,
+                sender: transferIntent.sender,
+                recipient: transferIntent.recipient,
+                blockTimestamp: transferIntent.blockTimestamp
+            }),
+            payment,
+            useQuotecall
+        );
+
+        ActionIntent memory actionIntent;
+        {
+            uint256[] memory amountOuts = new uint256[](1);
+            amountOuts[0] = transferIntent.amount;
+            string[] memory assetSymbolOuts = new string[](1);
+            assetSymbolOuts[0] = transferIntent.assetSymbol;
+            uint256[] memory amountIns = new uint256[](0);
+            string[] memory assetSymbolIns = new string[](0);
+            actionIntent = ActionIntent({
+                actor: transferIntent.sender,
+                amountIns: amountIns,
+                assetSymbolIns: assetSymbolIns,
+                amountOuts: amountOuts,
+                assetSymbolOuts: assetSymbolOuts,
+                blockTimestamp: transferIntent.blockTimestamp,
+                chainId: transferIntent.chainId,
+                useQuotecall: useQuotecall,
+                bridgeEnabled: true,
+                autoWrapperEnabled: true
+            });
+        }
+
+        (IQuarkWallet.QuarkOperation[] memory quarkOperationsArray, Actions.Action[] memory actionsArray) =
+        collectAssetsForAction({
+            actionIntent: actionIntent,
+            chainAccountsList: chainAccountsList,
+            payment: payment,
+            quarkOperation: operation,
+            action: action
+        });
+        return BuilderResult({
```

Comment:
Nit: add a newline before the `return` statement.

Reply:
Added.  👍